itwm_example/mco/mco.py

```diff
@@ -5,86 +5,45 @@
 from scipy import optimize
 
 from traits.api import (
-    HasStrictTraits, List, Float, Instance
+    HasStrictTraits,
+    List,
+    Float,
+    Instance,
+    Interface,
+    provides,
+    Type,
 )
 
-from force_bdss.api import (
-    BaseMCO, BaseMCOParameter, DataValue
-)
+from force_bdss.api import BaseMCO, BaseMCOParameter, DataValue
 from force_bdss.mco.i_evaluator import IEvaluator
 
 from .subprocess_workflow_evaluator import SubprocessWorkflowEvaluator
+from .scaling_tools.kpi_scaling import sen_scaling_method
 
 log = logging.getLogger(__name__)
 
 
-class MCO(BaseMCO):
-
-    def run(self, evaluator):
-
-        model = evaluator.mco_model
-        parameters = model.parameters
-        kpis = model.kpis
-
-        if model.evaluation_mode == "Subprocess":
-            # Here we create an instance of our WorkflowEvaluator subclass
-            # that allows for evaluation of a state in the workflow via calling
-            # force_bdss on a new subprocess running in 'evaluate' mode.
-            # Note: a BaseMCOCommunicator must be present to pass in parameter
-            # values and returning the KPI for a force_bdss run in 'evaluate'
-            # mode
-            single_point_evaluator = SubprocessWorkflowEvaluator(
-                workflow=evaluator.workflow,
-                workflow_filepath=evaluator.workflow_filepath,
-                executable_path=sys.argv[0]
-            )
-        else:
-            single_point_evaluator = evaluator
-
-        #: Get scaling factors and non-zero weight combinations for each KPI
-        scaling_factors = get_scaling_factors(single_point_evaluator,
-                                              kpis,
-                                              parameters)
-        weight_combinations = get_weight_combinations(len(kpis),
-                                                      model.num_points,
-                                                      False)
+class IOptimizer(Interface):
```

Comment:
A good idea could be to refactor out the `IOptimizer` classes into another module, maybe `itwm_example/mco/optimizers.py` for now. The same could also occur for any optimiser unit tests in `test_mco.py`.